pkg/storage/bucket/s3/bucket_client.go

```diff
@@ -37,7 +40,9 @@ func newS3Config(cfg Config) (s3.Config, error) {
 	if err != nil {
 		return s3.Config{}, err
 	}
-
+	if strings.HasPrefix(cfg.Endpoint, cfg.BucketName) {
+		return s3.Config{}, fmt.Errorf("the endpoint must not prefixed with the bucket name: %s", cfg.BucketName)
+	}
```

Comment:
We should do this in the `Config.Validate()` method:

https://github.com/grafana/mimir/blob/931429b092df580136a943f156d2429943c478b6/pkg/storage/bucket/s3/config.go#L104-L108

Reply:
@colega Thanks, reviewing. I fixed it and added change log and unit test.